File: ima-backend/src/main/java/com/techsolution/ima_backend/services/InternshipService.java
package com.techsolution.ima_backend.services;

import com.techsolution.ima_backend.dtos.request.InternshipRequest;
import com.techsolution.ima_backend.dtos.response.InternshipResponse;

import java.util.List;

public interface InternshipService {

    InternshipResponse createInternship(InternshipRequest internshipRequest);
    List<InternshipResponse> findAllActiveInternships();
    InternshipResponse findInternshipById(Long internshipId);
    InternshipResponse updateInternship(Long internshipId, InternshipRequest internshipRequest);
    void deactivateInternship(Long internshipId);
}

File: ima-frontend-react/src/services/InternshipService.js
import securedAxiosInstance from "./SecuredAxiosInstance";

const REST_API_INTERNSHIPS = '/internships';

export const listInternships = () => securedAxiosInstance.get(`${REST_API_INTERNSHIPS}`);

export const getInternship = (internshipId) => 
    securedAxiosInstance.get(`${REST_API_INTERNSHIPS}/${internshipId}`);

export const createInternship = (internshipRequest) => 
    securedAxiosInstance.post(`${REST_API_INTERNSHIPS}`, internshipRequest);

export const updateInternship = (internshipId, internshipRequest) => 
    securedAxiosInstance.put(`${REST_API_INTERNSHIPS}/${internshipId}`, internshipRequest);

export const deactivateInternship = (internshipId) => 
    securedAxiosInstance.delete(`${REST_API_INTERNSHIPS}/${internshipId}`);


